lightning/src/ln/onchaintx.rs

```diff
@@ -925,12 +914,9 @@ impl<ChanSigner: ChannelKeys> OnchainTxHandler<ChanSigner> {
 	// to monitor before.
 	pub(crate) fn get_fully_signed_holder_tx(&mut self, funding_redeemscript: &Script) -> Option<Transaction> {
 		if let Some(ref mut holder_commitment) = self.holder_commitment {
-			match self.key_storage.sign_holder_commitment(&holder_commitment, &self.secp_ctx) {
-				Ok(sig) => {
-					Some(holder_commitment.add_holder_sig(funding_redeemscript, sig))
-				},
-				Err(_) => return None,
-			}
+			let (sig, htlc_sigs) = self.key_storage.sign_holder_commitment(holder_commitment, &self.secp_ctx).expect("signing holder commitment");
+			self.holder_htlc_sigs = Some(Self::extract_holder_sigs(holder_commitment, htlc_sigs));
+			Some(holder_commitment.add_holder_sig(funding_redeemscript, sig))
 		} else {
 			None
```

Comment:
The function comment indicates that `self.holder_commitment` can be validly `None`.  Do you think the comment is wrong?

Reply:
Hmm, oops, but also isn't that out of date since 5d0bfa383425ecbf308fdf819dbf7ad19e6c5ca4? The comment describes the time between `get_outbound_funding_created` and `funding_signed` and since 80055d4bb41b83428f74141c0140165add2609a7 we remove some of the `Option`s which can cause it.

Reply:
nice catch!  I de-optionated things in a new commit.